packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -9,79 +9,153 @@ import 'package:file/file.dart';
     this.basedir, {
     this.fs = const LocalFileSystem(),
-  });
+    this.platform = const LocalPlatform(),
+  }) : assert(basedir != null);
 
-  /// The directory to which golden file URIs will be resolved in [compare] and [update].
+  /// The directory to which golden file URIs will be resolved in [compare] and
+  /// [update].
   final Uri basedir;
 
   /// The file system used to perform file access.
   @visibleForTesting
   final FileSystem fs;
 
-  /// Creates a new [FlutterGoldenFileComparator] that mirrors the relative
-  /// path resolution of the default [goldenFileComparator].
+  /// A wrapper for the [dart:io.Platform] API.
+  @visibleForTesting
+  final Platform platform;
+
+  @override
+  @protected
+  @visibleForTesting
+  Future<void> update(Uri golden, Uint8List imageBytes) async {
+    final File goldenFile = getGoldenFile(golden);
+    await goldenFile.parent.create(recursive: true);
+    await goldenFile.writeAsBytes(imageBytes, flush: true);
+  }
+
+  /// Modifies the given [Uri] to incorporate the current test library as a
+  /// prefix in the golden file name.
+  @override
+  @protected
+  @visibleForTesting
+  Uri getTestUri(Uri key, int version) {
+    return key;
+  }
+
+  /// Calculate the appropriate basedir for the current test context.
+  @protected
+  @visibleForTesting
+  static Directory getBaseDirectory(GoldensClient goldens, LocalFileComparator defaultComparator) {
+    final FileSystem fs = goldens.fs;
+    final Directory testDirectory = fs.directory(defaultComparator.basedir);
+    final String testDirectoryRelativePath = fs.path.relative(testDirectory.path, from: goldens.flutterRoot.path);
+    return goldens.comparisonRoot.childDirectory(testDirectoryRelativePath);
+  }
+
+  /// Returns the golden [File] identified by the given [Uri].
+  @protected
+  File getGoldenFile(Uri uri){
+    return fs.directory(basedir).childFile(fs.file(uri).path);
+  }
+}
+
+/// A [FlutterGoldenFileComparator] for testing golden images against the
+/// `flutter/goldens` repository.
+///
+/// Within the https://github.com/flutter/flutter repository, it's important
+/// not to check-in binaries in order to keep the size of the repository to a
+/// minimum. To satisfy this requirement, this comparator retrieves the golden
+/// files from a sibling repository, `flutter/goldens`.
+///
+/// This comparator will locally clone the `flutter/goldens` repository into
+/// the `$FLUTTER_ROOT/bin/cache/pkg/goldens` folder using the
+/// [GoldensRepositoryClient], then perform the comparison against the files
+/// therein.
+///
+/// See also:
+///
+///  * [GoldenFileComparator], the abstract class that
+///    [FlutterGoldenFileComparator] implements.
+///  * [FlutterSkiaGoldFileComparator], another [FlutterGoldenFileComparator]
+///    that tests golden images through Skia Gold.
+class FlutterGoldensRepositoryFileComparator extends FlutterGoldenFileComparator {
+  FlutterGoldensRepositoryFileComparator(Uri basedir) : super(basedir);
```

Comment:
Can you remove the test constructor and do here:

FlutterGoldensRepositoryFileComparator(Uri basedir, {FileSystem fs = const LocalFileSyetem, ...}) : super(basedir, fs: fs, ...);